job/job.go

```diff
@@ -41,7 +41,7 @@ type Job struct {
 // hwID is the hardware/worker ID corresponding to the MAC
 func hasActiveWorkflow(wcl *tw.WorkflowContextList) (bool, error) {
 	for _, wf := range (*wcl).WorkflowContexts {
-		if wf.CurrentActionState == tw.State_STATE_PENDING {
+		if wf.CurrentActionState == tw.State_STATE_PENDING || wf.CurrentActionState == tw.State_STATE_RUNNING {
```

Comment:
@splaspood objective of keeping only `Pending` workflow was to only pxe boot if there is a workflow which can be executed. Since a running workflow (already executed workflow) should not be executed again, it should be kept only `Pending`, instead, a new workflow should be created.

Reply:
IMHO this change is not required.

Reply:
We utilize hasActiveWorkflow to determine if we should accept (and pass on) 'status' messages to our central API.   Without this change, all those messages sent during the execution of a workflow are dropped due to the workflow not being considered to be 'active'.

Reply:
okay, in that case, I would suggest we should add a separate method to fetch running workflows similar to `hasActiveWorkflow` one, else this can affect pxe boot workflow. what do you think?

Reply:
I wonder if PXE booting when a workflow is RUNNING is really a problem.   Certainly we have no mechanisms in place currently to 'resume' a workflow execution post reboot (although, that sounds like something we'll want to consider/implement for unrelated reasons).   So if a workflow is RUNNING and the machine gets rebooted, it will PXE boot back into tink-worker, which will then likely fail in some manner.   Without this change, instead the machine would be rebooted, fail to PXE, and still fail to complete the RUNNING workflow.

Reply:
IIRC we always envisioned for tink-worker to be pxe booted to even if there's a workflow already runnning. tink-worker executions should not be arbitrarily blocked, especially if workflow actions are idempotent an interrupted action will be able to pick up and re do its bit.

There can be the case of wanting to execute actions from within an installed machine that might have trouble if the machine is always set to pxe boot. But that seems like a fringe issue. There's always kexec option and possibly using pbnj to ensure pxe boot is disabled.

Reply:
@Cbkhare going to dismiss this. This is the way we want to go in general. If you see a case where we don't want to netboot please open up a separate issue.